components/centre_power.py

```diff
@@ -30,13 +30,19 @@ def draw_base(self, canvas: Canvas):
         pass
 
     def draw_data(self, canvas: Canvas, data: Data):
-        power = data["power"]
-        rec_power = data["rec_power"]
+        power = data["power"].get_string()
+        rec_power = data["rec_power"].get_string()
 
-        power_str = f"{power:.0f}w"
-        rec_power_str = f"{rec_power:.0f}w rec"
+        if power is not None:
+            power_str = power + "w"
+        else:
+            power_str = "--"
```

Comment:
Since we can't do that, this is I reckon the next best thing.
```python
if data["power"].is_valid():
    power_str = data["power"].get_string() + "w"
else
    power_str = "--"
```
By doing this, we never create the potentially invalid variable `power`, which could be misused later if someone didn't know that it *could* be `None`

Reply:
Excuse me. Please use this by it's more formal terminology. It's called the [walrus operator](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions). Thank you for listening to my TED talk.

Reply:
I can't believe I forgot about this!